### PR TITLE
device.auto

### PR DESCRIPTION
**Pull request by j-proc**
## Description

The auto function now returns a CPU or GPU device object rather than existing as it's own class. 

## Motivation and Context

Resolves: #717 
Closes: #747

### CODE REVIEW

**Comment by j-proc** [comment]
Was able to just do very small changes, so I based it on your refactor branch and should be trivial to merge in now.

File: hoomd/device.py [hunk 1]
```diff
@@ -359,28 +359,29 @@ def __init__(self,
             self.num_cpu_threads = num_cpu_threads
 
 
-class Auto(_Device):
+def auto(communicator=None, msg_file=None, shared_msg_file=None, notice_level=2):
```
Comment (joaander) [inline hoomd/device.py:362]: Let's make the method a verb. `auto_select`?

File: hoomd/device.py [hunk 2]
```diff
@@ -359,28 +359,29 @@ def __init__(self,
...
     """
     Allow simulation hardware to be chosen automatically by HOOMD-blue
```
Comment (joaander) [inline hoomd/device.py:364]: Fix the pydocstyle errors:
```
device.py:363 in public function `auto`:
        D212: Multi-line docstring summary should start at the first line
device.py:363 in public function `auto`:
        D415: First line should end with a period, question mark, or exclamation point (not 'e')
```

File: hoomd/device.py (hunk 1 above)
Comment (j-proc) [reply hoomd/device.py]: Done

File: hoomd/device.py (hunk 2 above)
Comment (j-proc) [reply hoomd/device.py]: Done

**Pull request merged**